webpack.config.js

```diff
@@ -64,6 +65,35 @@ const stylesTransform = ( content ) => {
 	return content;
 };
 
+const blockNameRegex = new RegExp( /(?<=src\/).*(?=(\/frontend))/g );
+
+const createEntrypoints = () => {
+	const scriptPaths = fastGlob.sync(
+		'./packages/block-library/src/**/frontend.js',
+		{
+			ignore: [ '**/build*/**' ],
+		}
+	);
+
+	const scriptEntries = scriptPaths.reduce( ( entries, scriptPath ) => {
+		const [ blockName ] = scriptPath.match( blockNameRegex ) || [];
```

Comment:
What will happen when the block name is undefined?

Reply:
`fast-glob` will look for `frontend.js` files is within block source directories, so the value of `scriptPaths` will always be either an empty array, or a list of paths in the form `'./packages/block-library/src/<blockName>/frontend.js'`, in which case `/(?<=src\/).*(?=(\/frontend))/g` would always have a match.

Because of this, the guard I set there is unnecessary. 

In order to make sure, I ran `npm run build`, without the guard present, under the following conditions:
- no `frontend.js` files present.
- `frontend.js` file present in a subdirectory of a block's source directory, in which case the file gets built into that same subdirectory in the block's build directory
- `frontend.js` present in multiple block directories. 

I think the best think to do here would be deleting the misleading guard.
I've done this and also added more comments going through the logic in https://github.com/WordPress/gutenberg/pull/30629

Reply:
Got it, makes total sense, this `[]` was confusing 😄